packages/toc/src/toc_item.tsx

```diff
@@ -2,14 +2,75 @@
 // Distributed under the terms of the Modified BSD License.
 
 import * as React from 'react';
-import { IHeading } from './utils/headings';
+import { IHeading, INotebookHeading } from './utils/headings';
+import ListItem from '@material-ui/core/ListItem';
+import Menu from '@material-ui/core/Menu';
+import MenuItem from '@material-ui/core/MenuItem';
```

Comment:
Thank you for picking up this work, it might be useful indeed. We should probably use [`lumino` Menu](https://jupyterlab.github.io/lumino/widgets/classes/menu.html) instead, at least until there is a consensus decision on https://github.com/jupyterlab/jupyterlab/issues/7574. This would avoid adding more dependencies, each of which may introduce a security issue. Apologies if it was already agreed and I missed it.